src/VclHandler.php

```diff
@@ -0,0 +1,739 @@
+      return true;
+    }
+    return false;
+
+  }
+
+  /**
+   * Prepares update setting data
+   *
+   * @data array
+   * @return array
+   */
+  public function prepare_update_setting($data) {
+    $url = $this->_versionBaseUrl . '/' . $this->_lastClonedVersion . '/request_settings/' . $data['name'];
+
+    $request = array(
+      'url' => $url,
+      'data' => $data,
+      'type' => 'PUT'
+    );
+
+    return $request;
+  }
+
+  /**
+   * Prepares Insert setting data
+   *
+   * @data array
+   * @return array
+   */
+  public function prepare_insert_setting($data) {
+    $url = $this->_versionBaseUrl . '/' . $this->_lastClonedVersion . '/request_settings';
+
+    $request = array(
+      'url' => $url,
+      'data' => $data,
+      'type' => 'POST'
+    );
+
+    return $request;
+  }
+
+  /**
+   * Validates last cloned version
+   *
+   * @return bool
+   */
+  public function validateVersion() {
+    $url = $this->_versionBaseUrl . '/' . $this->_lastClonedVersion . '/validate';
+    $response = $this->vclGetWrapper($url, $this->_headersGet);
+
+    if($response->getStatusCode() != "200") {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Activates last cloned version
+   *
+   * @return array
+   */
+  public function prepareActivateVersion() {
+    $url = $this->_versionBaseUrl . '/' . $this->_lastClonedVersion . '/activate';
+
+    $request = array(
+      'url' => $url,
+      'type' => 'PUT',
+      'headers' => $this->_headersGet
+    );
+
+    return $request;
+  }
+
+  /**
+   * Adds new error to error array
+   *
+   * @param string $message
+   */
+  public function addError($message) {
+    $this->_errors[] = $message;
+  }
+
+  /**
+   * Fetches logged errors
+   *
+   * @return array
+   */
+  public function getErrors() {
+    return $this->_errors;
+  }
+
+  /**
+   * Function for testing Fastly API connection
+   *
+   * @param $hostname
+   * @param $service_id
+   * @param $api_key
+   * @return array
+   */
+  public function testFastlyApiConnection($hostname, $service_id, $api_key) {
+    if (empty($hostname) || empty($service_id) || empty($api_key)) {
+      return array('status' => false, 'message' => 'Please enter credentials first');
+    }
+
+    $url = '/service/' . $service_id;
+    $headers = array(
+      'Fastly-Key' => $api_key,
+      'Accept' => 'application/json'
+    );
+    try {
+
+      $response = $this->vclGetWrapper($url, $headers);
+      if ($response->getStatusCode() == "200") {
+        $status = true;
+        $response_body = json_decode($response->getBody());
+        $service_name = $response_body->name;
+        $message = 'Connection Successful on service *' . $service_name . "*";
+      } else {
+        $status = false;
+        $response_body = json_decode($response->getBody());
+        $service_name = $response_body->name;
+        $message = 'Connection not Successful on service *' . $service_name . "*" . " status : " . $response->getStatusCode();
```

Comment:
Look at using placeholder replacement in the log message:

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Render%21FormattableMarkup.php/function/FormattableMarkup%3A%3AplaceholderFormat/8.3.x

https://drupalize.me/blog/201510/how-log-messages-drupal-8